packages/core/src/sync-realtime/index.ts

```diff
@@ -134,14 +137,87 @@ export const createRealtimeSync = (
    * 1 block ahead of the local chain.
    * @returns true if a reorg occurred
    */
-  const handleBlock = async (block: SyncBlock) => {
+  const handleBlock = async ({
+    block,
+    logs,
+    callTraces,
+    transactions,
+    transactionReceipts,
+  }: Omit<BlockWithEventData, "filters">) => {
     args.common.logger.debug({
       service: "realtime",
       msg: `Started syncing '${args.network.name}' block ${hexToNumber(block.number)}`,
     });
 
-    const { logs, callTraces, transactions, transactionReceipts, filters } =
-      await extract(block);
+    // Get all addresses from `logs` that are from a factory
+    const childAddresses = new Set<Address>();
+    for (const filter of factories) {
+      if (logs.length > 0) {
+        const _childAddresses = await args.syncStore.filterChildAddresses({
+          filter,
+          addresses: logs.map(({ address }) => address),
+        });
+        for (const address of _childAddresses) childAddresses.add(address);
+      }
+    }
+
+    /**
+     * `logs` and `callTraces` must be filtered again (already filtered in `extract`)
+     *  because `extract` doesn't have factory address information.
+     */
+
+    const matchedFilters = new Set<Filter>();
+
+    // Remove logs that don't match a filter, accounting for factory addresses
+    logs = logs.filter((log) => {
+      const isFilterMatched = logFilters.map((filter) => {
+        const isMatched = isLogFilterMatched({ filter, block, log });
+
+        if (isAddressFactory(filter.address)) {
+          return isMatched && childAddresses.has(log.address);
+        }
+
+        return isMatched;
+      });
+
+      for (let i = 0; i < logFilters.length; i++) {
+        if (isFilterMatched[i]!) matchedFilters.add(logFilters[i]!);
+      }
+
+      return isFilterMatched.some((m) => m);
+    });
+
+    // Remove call traces that don't match a filter, accounting for factory addresses
+    callTraces = callTraces.filter((callTrace) => {
```

Comment:
Confirming my understanding here - is this logic equivalent?
```ts
callTraces = callTraces.filter((callTrace) => {
  let isMatched = false;
  
  for (const filter of callTraceFilters) {
    if (
      isCallTraceFilterMatched({ filter, block, callTrace }) &&
      (isAddressFactory(filter.toAddress)
        ? childAddresses.has(callTrace.action.to)
        : true)
    ) {
      matchedFilters.add(filter);
      isMatched = true;
    }
  }
  
  return isMatched;
});
```

Reply:
yes